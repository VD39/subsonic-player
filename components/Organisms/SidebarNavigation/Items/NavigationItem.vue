<script setup lang="ts">
defineProps<{
  collapsed: boolean;
  title: string;
}>();
</script>

<template>
  <li>
    <h4
      ref="title"
      :class="[
        'strong',
        'mBM',
        'sidebarTitle',
        {
          visuallyHidden: collapsed,
        },
      ]"
    >
      {{ title }}
    </h4>

    <ul v-if="$slots.default" ref="navigation">
      <slot />
    </ul>
  </li>
</template>
