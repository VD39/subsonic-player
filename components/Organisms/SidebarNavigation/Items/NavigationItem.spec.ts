import type { VueWrapper } from '@vue/test-utils';

import { mount } from '@vue/test-utils';

import NavigationItem from './NavigationItem.vue';

function factory(props = {}, slots = {}) {
  return mount(NavigationItem, {
    props: {
      collapsed: false,
      title: 'title',
      ...props,
    },
    slots: {
      ...slots,
    },
  });
}

describe('NavigationItem', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = factory();
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('when collapsed prop is false', () => {
    it('does not add the visuallyHidden class to the title element', () => {
      expect(wrapper.find({ ref: 'title' }).classes()).not.toContain(
        'visuallyHidden',
      );
    });
  });

  describe('when collapsed prop is true', () => {
    beforeEach(() => {
      wrapper = factory({
        collapsed: true,
      });
    });

    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('adds the visuallyHidden class to the title element', () => {
      expect(wrapper.find({ ref: 'title' }).classes()).toContain(
        'visuallyHidden',
      );
    });
  });

  describe('when slot data is not added', () => {
    it('does not show the navigation element', () => {
      expect(wrapper.find({ ref: 'navigation' }).exists()).toBe(false);
    });
  });

  describe('when slot data is added', () => {
    beforeEach(() => {
      wrapper = factory(undefined, {
        default: 'Default slot content.',
      });
    });

    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('shows the navigation element', () => {
      expect(wrapper.find({ ref: 'navigation' }).exists()).toBe(true);
    });
  });
});
