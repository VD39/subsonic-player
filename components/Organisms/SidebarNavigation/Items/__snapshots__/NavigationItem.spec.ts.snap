// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationItem > matches the snapshot 1`] = `
"<li>
  <h4 class="strong mBM sidebarTitle">title</h4>
  <!--v-if-->
</li>"
`;

exports[`NavigationItem > when collapsed prop is true > matches the snapshot 1`] = `
"<li>
  <h4 class="strong mBM sidebarTitle visuallyHidden">title</h4>
  <!--v-if-->
</li>"
`;

exports[`NavigationItem > when slot data is added > matches the snapshot 1`] = `
"<li>
  <h4 class="strong mBM sidebarTitle">title</h4>
  <ul>Default slot content.</ul>
</li>"
`;
